common/libs/validators.js

```diff
@@ -84,3 +84,45 @@ export function isPositiveIntegerOrZero(number: number): boolean {
   }
   return number >= 0 && parseInt(number) === number;
 }
+
+export function isValidRawTx(rawTx: Object): boolean {
+  const propReqs = [
+    { name: 'nonce', type: 'string', lenReq: true },
+    { name: 'gasPrice', type: 'string', lenReq: true },
+    { name: 'gasLimit', type: 'string', lenReq: true },
+    { name: 'to', type: 'string', lenReq: true },
+    { name: 'value', type: 'string', lenReq: true },
+    { name: 'data', type: 'string', lenReq: false },
+    { name: 'chainId', type: 'number' }
+  ];
+  let valid = true;
+
+  //ensure rawTx has above properties
+  //ensure all specified types match
+  //ensure length !0 for strings where length is required
+  //ensure valid hex for strings
+  //ensure all strings begin with '0x'
+  //ensure valid address for 'to' prop
+  //ensure rawTx only has above properties
+
+  propReqs.forEach(prop => {
+    if (!valid) return;
+
+    const value = rawTx[prop.name];
+
+    if (!rawTx.hasOwnProperty(prop.name)) return (valid = false);
```

Comment:
Honestly this seems like a good use-case for the good ol' fashioned for loop, to spare dealing with the `valid` variable and just return out of the function at the first sign of badness.